infrastructures/infrastructure-factory/src/main/java/org/eclipse/che/api/factory/server/scm/kubernetes/KubernetesAuthorisationRequestManager.java

```diff
@@ -0,0 +1,145 @@
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Red Hat, Inc. - initial API and implementation
+ */
+package org.eclipse.che.api.factory.server.scm.kubernetes;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+import static org.eclipse.che.commons.lang.UrlUtils.getParameter;
+import static org.eclipse.che.commons.lang.UrlUtils.getQueryParametersFromState;
+import static org.eclipse.che.commons.lang.UrlUtils.getRequestUrl;
+import static org.eclipse.che.commons.lang.UrlUtils.getState;
+import static org.eclipse.che.workspace.infrastructure.kubernetes.namespace.AbstractWorkspaceServiceAccount.PREFERENCES_CONFIGMAP_NAME;
+
+import com.google.gson.Gson;
+import io.fabric8.kubernetes.api.model.ConfigMap;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import jakarta.ws.rs.core.UriInfo;
+import java.net.URL;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.eclipse.che.api.factory.server.scm.exception.ScmConfigurationPersistenceException;
+import org.eclipse.che.api.factory.server.scm.exception.UnsatisfiedScmPreconditionException;
+import org.eclipse.che.api.workspace.server.spi.InfrastructureException;
+import org.eclipse.che.security.oauth.AuthorisationRequestManager;
+import org.eclipse.che.workspace.infrastructure.kubernetes.CheServerKubernetesClientFactory;
+import org.eclipse.che.workspace.infrastructure.kubernetes.api.shared.KubernetesNamespaceMeta;
+import org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesNamespaceFactory;
+
+/** Store and retrieve rejected authorisation requests in the Kubernetes ConfigMap. */
+@Singleton
+public class KubernetesAuthorisationRequestManager implements AuthorisationRequestManager {
+  private final KubernetesNamespaceFactory namespaceFactory;
+  private final CheServerKubernetesClientFactory cheServerKubernetesClientFactory;
+  private static final String SKIP_AUTHORISATION_MAP_KEY = "skip-authorisation";
+
+  @Inject
+  public KubernetesAuthorisationRequestManager(
+      KubernetesNamespaceFactory namespaceFactory,
+      CheServerKubernetesClientFactory cheServerKubernetesClientFactory) {
+    this.namespaceFactory = namespaceFactory;
+    this.cheServerKubernetesClientFactory = cheServerKubernetesClientFactory;
+  }
+
+  @Override
+  public void store(String scmProviderName) {
+    if (isStored(scmProviderName)) {
+      return;
+    }
+    ConfigMap configMap = getConfigMap();
+    HashSet<String> fromJson = getSkipAuthorisationValues();
+    fromJson.add(scmProviderName);
+
+    configMap.setData(Map.of(SKIP_AUTHORISATION_MAP_KEY, fromJson.toString()));
+
+    patchConfigMap(configMap);
+  }
+
+  @Override
+  public void remove(String scmProviderName) {
```

Comment:
If the user denies the OAuth request, is there any flow for "I changed my mind"? It seems like with this change Che will silently never ask again for OAuth access. This is probably fine for now, but maybe something we should think about in the future.

Reply:
This is going to be solved by the next iteration in the dashboard part. We are going to have an additional checkbox, or something like that, for each item in the `Git Services` tab of the user preferences page.

Reply:
The corresponding dashboard issue has been created https://github.com/eclipse/che/issues/22600